packages/@react-aria/select/src/HiddenSelect.tsx

```diff
@@ -99,7 +100,7 @@ export function useHiddenSelect<T>(props: AriaHiddenSelectOptions, state: Select
       disabled: isDisabled,
       required: validationBehavior === 'native' && isRequired,
       name,
-      value: state.selectedKey ?? '',
+      value: state.selectedKey ?? (selectRefValue || ''),
```

Comment:
I propose this instead
```suggestion
      value: state.selectedKey ?? undefined,
```
I'm not sure how I'm avoiding the uncontrolled -> controlled issue, but I get no warnings for it.